# Title of Project

Financial Market News Sentiment Analysis

## **Objectives**

In [6]:
To analyze Financial Market News based on sentiment

SyntaxError: invalid syntax (44105147.py, line 1)

# Data Source

 From YBI Foundation GitHub we take this dataset.

# Import library

In [ ]:
import pandas as pd
import numpy as np

# Import dataset

In [ ]:
df = pd.read_csv(r'https://raw.githubusercontent.com/YBI-Foundation/Dataset/main/Financial%20Market%20News.csv',  encoding = "ISO-8859-1")

In [ ]:
df.head()

# Describe Data

In [ ]:
df.info()

In [ ]:
df.shape

In [7]:
df.columns

Index(['Date', 'Label', 'News 1', 'News 2', 'News 3', 'News 4', 'News 5',
       'News 6', 'News 7', 'News 8', 'News 9', 'News 10', 'News 11', 'News 12',
       'News 13', 'News 14', 'News 15', 'News 16', 'News 17', 'News 18',
       'News 19', 'News 20', 'News 21', 'News 22', 'News 23', 'News 24',
       'News 25'],
      dtype='object')

# Data Preprocessing

# Get Feature Selection

In [8]:
' '.join(str(x) for x in df.iloc[1,2:27])

"Warning from history points to crash Investors flee to dollar haven Banks and tobacco in favour Review: Llama Farmers War jitters lead to sell-off Your not-so-secret history Review: The Northern Sinfonia Review: Hysteria Review: The Guardsman Opera: The Marriage of Figaro Review: The Turk in Italy Deutsche spells out its plans for diversification Traders' panic sends oil prices skyward TV sport chief leaves home over romance Leader: Hi-tech twitch Why Wenger will stick to his Gunners Out of luck England hit rock bottom Wilkinson out of his depth Kinsella sparks Irish power play Brown banished as Scots rebound Battling Wales cling to lifeline Ehiogu close to sealing Boro move Man-to-man marking Match stats French referee at centre of storm is no stranger to controversy"

In [9]:
df.index

RangeIndex(start=0, stop=4101, step=1)

In [10]:
len(df.index)

4101

In [11]:
news = []
for row in range(0,len(df.index)):
  news.append(''.join(str(x) for x in df.iloc[row, 2:27]))

In [12]:
type(news)

list

In [13]:
news[0]

"McIlroy's men catch cold from GudjonssonObituary: Brian WalshWorkplace blues leave employers in the redClassical review: RattleDance review: Merce CunninghamGenetic tests to be used in setting premiumsOpera review: La BohèmePop review: Britney SpearsTheatre review: The CircleWales face a fraught nightUnder-21  round-upSmith off to blot his copybookFinns taking the mickeyPraise wasted as Brown studies injury optionsIreland wary of minnowsFinland 0 - 0 EnglandHealy a marked manHappy birthday Harpers & QueenWin unlimited access to the Raindance film festivalLabour pledges £800m to bridge north-south divideWales: Lib-Lab pact firm despite resignationDonald DewarRegenerating homes  regenerates well-being in peopleWin £100 worth of underwearTV guide: Random views"

In [14]:
X = news

In [15]:
type(X)

list

# Get Features Text Conversation to Bag of words

In [16]:
from sklearn.feature_extraction.text import CountVectorizer

In [17]:
cv = CountVectorizer(lowercase= True,ngram_range=(1,1))

In [18]:
X = cv.fit_transform(X)

In [19]:
X.shape

(4101, 108682)

In [20]:
y = df ['Label']

In [31]:
y.shape

(4101,)

# Get train Test split

In [21]:
from sklearn.model_selection import train_test_split

In [22]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, stratify = y, random_state = 2529)

In [23]:
from sklearn.ensemble import RandomForestClassifier

In [24]:
rf = RandomForestClassifier(n_estimators=200)

In [25]:
rf.fit(X_train ,y_train)

RandomForestClassifier(n_estimators=200)

In [26]:
y_pred = rf.predict(X_test)

In [27]:
from sklearn.metrics import classification_report,confusion_matrix,accuracy_score

In [28]:
confusion_matrix(y_test, y_pred)

array([[108, 473],
       [118, 532]], dtype=int64)

In [29]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.48      0.19      0.27       581
           1       0.53      0.82      0.64       650

    accuracy                           0.52      1231
   macro avg       0.50      0.50      0.46      1231
weighted avg       0.51      0.52      0.47      1231



# Explanation

52% Accuracy was achieved using Random Forest Classifier